### PR TITLE
[Feature] Add PaddleFormers fallback backend

### PR DESCRIPTION
**Pull request by jackyYang6**
## Motivation

When FastDeploy lacks a native implementation for a specific model architecture, users must wait for official support or implement it themselves. This PR introduces a PaddleFormers fallback mechanism that allows FastDeploy to quickly support any PaddleFormers-compatible model by reusing PaddleFormers model definitions while retaining FastDeploy's scheduling and serving capabilities.

### New Module: `fastdeploy/model_executor/models/paddleformers/`

| File | Description |
| --- | --- |
| `__init__.py` | Registers PaddleFormersForCausalLM with ModelRegistry |
| `base.py` | Base model class: recursive layer replacement, weight loading, forward pass |
| `causallm.py` | CausalLM Mixin: lm_head initialization, logits computation |

### Public Module Adaptations

| File | Changes |
| --- | --- |
| `config.py` | Added `model_impl` config option (auto/fastdeploy/paddleformers) |
| `args_utils.py` | Added `--model-impl` CLI argument |
| `model_base.py` | Added fallback resolution logic in `_try_resolve_paddleformers` |
| `forward_meta.py` | Added `rope_already_applied` flag to prevent double RoPE application |
| `append_attn_backend.py` | Added identity RoPE support for fallback models |
| `gpu_model_runner.py` | Changed to kwargs-style model invocation |

### Online Mode (API Server)

```bash
# Auto mode: use native implementation if available, fallback to PaddleFormers otherwise
python -m fastdeploy.entrypoints.openai.api_server \
    --model Qwen/Qwen3-4B \
    --model-impl auto

# Force PaddleFormers backend
python -m fastdeploy.entrypoints.openai.api_server \
    --model Qwen/Qwen3-4B \
    --model-impl paddleformers
```
### Offline Mode (LLM Engine)
```python
from fastdeploy import LLM

llm = LLM(model="Qwen/Qwen3-4B", model_impl="auto")
llm = LLM(model="Qwen/Qwen3-4B", model_impl="paddleformers")
``` 

### Parameter Reference
| Value | Behavior |
| --- | --- |
| `auto` (default) | Use native FastDeploy implementation when available; fallback to PaddleFormers otherwise |
| `fastdeploy` | Use native implementation only; raise error if unavailable |
| `paddleformers` | Force PaddleFormers backend |

## Checklist

- [ ] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [ ] Format your code, run `pre-commit` before commit.
- [ ] Add unit tests. Please write the reason in this PR if no unit tests.
- [ ] Provide accuracy results.
- [ ] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: docs/zh/features/paddleformers_backend.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# PaddleFormers 后端
```
Comment (Jiang-Jia-Jun) [inline docs/zh/features/paddleformers_backend.md:1]: 添加英文

File: fastdeploy/config.py [hunk 2]
```diff
@@ -314,6 +321,7 @@ def _post_init(self):
         model_info, arch = registry.inspect_model_cls(self.architectures, self)
         self._model_info = model_info
         self._architecture = arch
+        self.architectures = [arch]
```
Comment (Jiang-Jia-Jun) [inline fastdeploy/config.py:324]: 多增加的这个变量的目的是什么，为什么不复用self._architecture呢

**Review by Jiang-Jia-Jun — DISMISSED** [review]
